blockchain/blockchain.go

```diff
@@ -658,6 +658,9 @@ func (bc *BlockChain) FastSyncCommitHead(hash common.Hash) error {
 // CurrentBlock retrieves the current head block of the canonical chain. The
 // block is retrieved from the blockchain's internal cache.
 func (bc *BlockChain) CurrentBlock() *types.Block {
+	if bc.db.GetDBConfig().DBType == database.RocksDB && bc.db.GetDBConfig().RocksDBConfig.Secondary {
+		bc.loadLastState()
```

Comment:
Tell me if I understood - secondary nodes do not insert blocks themselves; so they are not notified of new blocks. Thus `bc.currentBlock` is kept untouched. To know the current state in a timely manner, every `CurrentBlock` call queries the database.

If so, then why update here? I guess because `CurrentBlock()` is used in most of the RPCs? I somewhat agree with the choice. But code-wise it looks arbitrary that only CurrentBlock updates and other getters (such as GetBlock) do not.

How about having a background loop that periodically loadLastState, to not depend on a specific method's side effect.

Reply:
Yes, you are right. Still `CurrentBlock` is called periodically somewhere while processing blocks, but I agreed with your point. Let me find better place to update last state, or I will take your advice.

Reply:
Thank you.
Another concern occurs to me though. loadLastState is a little heavy task involving several database reads. Having to loadLastState on every getter call might add some DB pressure. Though I'm just speculating because I don't know much about RocksDB client-side caching or remote read performance.

Reply:
@blukat29 I have updated this. The logic is borrowed from KES logic. Instead of subscribing current block from redis-cache, this reads current block from rocksdb per 1 second (would be better parameterized). DB reads only twice for `HeadBlockHash` and `Block` per a second so that I think it would be fine. It replaces current block along with updating assoicated variables and metrics. Also, resetting txpool enables to handle transactions, too. However, it won't update subscription data (e.g. tx log) at the momoment.

Please take a look and leave your thought. Thanks.

Reply:
@jeongkyun-oh Looks good to me. Reading per 1 second (or any reasonable interval) will be good.
My previous concern was that reading per every CurrentBlock call would cause hundreds or thousands of reads per sec. Using the loop clears my concern.